systemtest/src/test/java/io/strimzi/systemtest/kafka/listeners/MultipleListenersST.java

```diff
@@ -0,0 +1,279 @@
+                    kafkaBasicClientJob.producerStrimzi().done();
+                    kafkaBasicClientJob.consumerStrimzi().done();
+
+                    // verify phase
+                    ClientUtils.waitForClientSuccess(producerName, NAMESPACE, MESSAGE_COUNT);
+                    ClientUtils.waitForClientSuccess(consumerName, NAMESPACE, MESSAGE_COUNT);
+
+
+                    LOGGER.info("Deleting the Jobs");
+                    // teardown (for clients)
+                    JobUtils.deleteJobWithWait(NAMESPACE, producerName);
+                    JobUtils.deleteJobWithWait(NAMESPACE, consumerName);
+                }
+            }
+        })));
+        return dynamicTests.iterator();
+    }
+
+    private List<List<GenericKafkaListener>> generateTestCases() {
+
+        List<List<GenericKafkaListener>> testCases = new ArrayList<>(10);
+
+        LOGGER.info("Starting to generate test cases for multiple listeners");
+
+        for (int i = 0; i < 10; i++) {
+
+            KafkaListenerType stochasticChosenListener = KafkaListenerType.values()[ThreadLocalRandom.current().nextInt(0, KafkaListenerType.values().length - 1)];
+            List<GenericKafkaListener> testCase = new ArrayList<>(5);
+            int stochasticCount;
+
+            switch (stochasticChosenListener) {
+                case NODEPORT:
+                    stochasticCount = ThreadLocalRandom.current().nextInt(2, 5);
+
+                    for (int j = 0; j < stochasticCount; j++) {
+
+                        boolean stochasticCommunication = ThreadLocalRandom.current().nextInt(2) == 0;
+
+                        testCase.add(new GenericKafkaListenerBuilder()
+                            .withName(generateRandomListenerName())
+                            .withPort(6090 + j)
+                            .withType(KafkaListenerType.NODEPORT)
+                            .withTls(stochasticCommunication)
+                            .build());
+                    }
+                    break;
+                case LOADBALANCER:
+                    stochasticCount = ThreadLocalRandom.current().nextInt(2, 3);
+
+                    for (int j = 0; j < stochasticCount; j++) {
+
+                        boolean stochasticCommunication = ThreadLocalRandom.current().nextInt(2) == 0;
+
+                        testCase.add(new GenericKafkaListenerBuilder()
+                            .withName(generateRandomListenerName())
+                            .withPort(7090 + j)
+                            .withType(KafkaListenerType.LOADBALANCER)
+                            .withTls(stochasticCommunication)
+                            .build());
+                    }
+                    break;
+                case ROUTE:
+                    stochasticCount = ThreadLocalRandom.current().nextInt(2, 3);
+
+                    for (int j = 0; j < stochasticCount; j++) {
+
+                        boolean stochasticCommunication = ThreadLocalRandom.current().nextInt(2) == 0;
```

Comment:
this will be removed because `Route or Ingress type listener and requires enabled TLS encryption`